app/src/main/java/tech/relaycorp/gateway/data/repos/ParcelRepository.kt

```diff
@@ -0,0 +1,27 @@
+package tech.relaycorp.gateway.data.repos
+
+import androidx.room.*
+import kotlinx.coroutines.flow.Flow
+import tech.relaycorp.gateway.data.model.MessageId
+import tech.relaycorp.gateway.data.model.PrivateMessageAddress
+import tech.relaycorp.gateway.data.model.StorageSize
+import tech.relaycorp.gateway.data.model.StoredParcel
+
+@Dao
+interface ParcelRepository {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(message: StoredParcel)
+
+    @Delete
+    suspend fun delete(message: StoredParcel)
+
+    @Query("SELECT * FROM Parcel")
+    fun observeAll(): Flow<List<StoredParcel>>
+
+    @Query("SELECT COALESCE(SUM(Parcel.size), 0) FROM Parcel")
+    fun observeTotalSize(): Flow<StorageSize>
```

Comment:
I guess methods like this will depend on the destination of the parcels. For example, in the screen where we should the size of the data to be sent to couriers, we'd want to exclude the data that hasn't yet been delivered to local endpoints.

Reply:
Right, this method was copied over, but it's not being used at the moment. I can remove it, and only add it when we have an actual usage for it.